src/view/components/microbit/MicrobitImage.tsx

```diff
@@ -3,8 +3,83 @@
 
 import * as React from "react";
 import "../../styles/Microbit.css";
-import { MICROBIT_SVG } from "./Microbit_svg";
+import { MicrobitSvg } from "./Microbit_svg";
 
-export const MicrobitImage: React.FC = () => {
-    return MICROBIT_SVG;
+interface EventTriggers {
+    onMouseUp: (button: HTMLElement, event: Event, buttonKey: string) => void;
+    onMouseDown: (button: HTMLElement, event: Event, buttonKey: string) => void;
+    onMouseLeave: (
+        button: HTMLElement,
+        event: Event,
+        buttonKey: string
+    ) => void;
+}
+interface IProps {
+    eventTriggers: EventTriggers;
+    leds: number[][];
+}
+interface IState {
+    microbitImageRef: React.RefObject<SVGSVGElement>;
+    buttonRefs: { [key: string]: React.RefObject<SVGRectElement> };
+}
+
+// Displays the SVG and call necessary svg modification.
+export class MicrobitImage extends React.Component<IProps, IState> {
+    private svgRef: React.RefObject<MicrobitSvg> = React.createRef();
+    constructor(props: IProps) {
+        super(props);
+    }
+    componentDidMount() {
+        const svgElement = this.svgRef.current;
+        if (svgElement) {
+            updateAllLeds(this.props.leds, svgElement.getLeds());
+            setupAllButtons(this.props.eventTriggers, svgElement.getButtons());
+        }
+    }
+    componentDidUpdate() {
+        if (this.svgRef.current) {

```

Comment:
null check missing this.svgRef

Reply:
same here, svgRef should always be defined